sqllogictest/src/runner.rs

```diff
@@ -691,14 +699,13 @@ impl<D: AsyncDB> Runner<D> {
                     }
                 }
 
-                // We compare normalized results. Whitespace characters are ignored.
-                let normalized_rows = rows
-                    .into_iter()
-                    .map(|strs| strs.iter().map(normalize_string).join(" "))
-                    .collect_vec();
-
-                let expected_results = expected_results.iter().map(normalize_string).collect_vec();
-                if !(self.validator)(&normalized_rows, &expected_results) {
+                if !(self.validator)(&rows, &expected_results) {
+                    let normalized_rows = rows
+                        .into_iter()
+                        .map(|strs| strs.iter().map(normalize_string).join(" "))
+                        .collect_vec();
+                    let expected_results =
+                        expected_results.iter().map(normalize_string).collect_vec();
```

Comment:
It might be better to use raw input in the Error message? 🤔

Reply:
In fact, I think should use strings in `Validator`. Or the Error message will be misleading due to some whitespaces.

Reply:
Do you mean that use the string after a conversion in `Validator`? It makes some sense but first currently `Validator` interface doesn't necessarily convert results to some string.. If you want that maybe it should be e.g.,
```
pub type Validator = (fn(&Vec<String>) -> String, fn(&String>) -> String); 
```
😇

Second I think raw output is still useful to know what happened. The Validator can produce some random stuff... (Although not very likely)

Reply:
> Do you mean using the string after a conversion in `Validator`?

Yes, but will change `Validator` type again and make it complex.

Let's use raw input.